online-shop/src/app/add-product/add-product.component.ts

```diff
@@ -19,36 +23,25 @@ export class AddProductComponent implements OnInit {
   })
 
   constructor(private fb: FormBuilder,
-              private productsService: ProductsService,
               private route: ActivatedRoute,
-              private router: Router) { }
+              private router: Router,
+              private store: Store<AppState>) { }
 
   ngOnInit(): void {
   }
 
-
   submitHandler() {
     const formValue = this.addProductForm.value;
 
-    const data = {
+    const data: ProductAdd = {
```

Comment:
Neat trick here would be to use the spread operator on formValue such that the properties will be pasted into your data object. E.g., `const data: ProductAdd = { ...formValue }`